plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/snowflake/IcebergSnowflakeCatalogConfig.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.snowflake;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.NotNull;
+import net.snowflake.client.jdbc.SnowflakeDriver;
+
+import java.net.URI;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.util.Optional;
+
+public class IcebergSnowflakeCatalogConfig
+{
+    private URI uri;
+    private String user;
+    private String password;
+    private String database;
+    private Optional<String> role = Optional.empty();
+
+    @AssertTrue(message = "Invalid JDBC URL for Iceberg Snowflake catalog")
```

Comment:
Could you test that the validation fails when using an invalid jdbc url on `TestIcebergSnowflakeCatalogConfig` ?

Reply:
Added